preload/link-header-preload.html

```diff
@@ -1,20 +1,33 @@
 <!DOCTYPE html>
+<title>Makes sure that Link headers preload resources</title>
 <script src="/resources/testharness.js"></script>
 <script src="/resources/testharnessreport.js"></script>
 <script src="/preload/resources/preload_helper.js"></script>
-<script>
-    var t = async_test('Makes sure that Link headers preload resources');
-</script>
 <body>
 <script>
-    window.addEventListener("load", t.step_func(function() {
-        t.step_timeout(function() {
-            verifyPreloadAndRTSupport();
-            verifyNumberOfResourceTimingEntries("resources/square.png?link-header-preload", 1);
-            verifyNumberOfResourceTimingEntries("resources/dummy.js?link-header-preload", 1);
-            verifyNumberOfResourceTimingEntries("resources/dummy.css?link-header-preload", 1);
-            t.done();
-        }, 5000);
-    }));
+setup({explicit_done: true});
+
+var iterations = 0;
+
+function check_finished() {
+  if (numberOfResourceTimingEntries("resources/square.png?link-header-preload") == 1 &&
+    numberOfResourceTimingEntries("resources/dummy.js?link-header-preload") == 1 &&
+    numberOfResourceTimingEntries("resources/dummy.css?link-header-preload") == 1) {
+    done();
+  }
+  iterations++;
+  if (iterations == 10) {
+    verifyNumberOfResourceTimingEntries("resources/square.png?link-header-preload", 1);
```

Comment:
Why do you need to verify here if you already checked for the values earlier? I don't think these asserts will ever pass. Or are you expecting them to fail? If so, maybe make that failure more explicit? (e.g. assert_unreached)

Reply:
I wanted to make this test behave similarly to the current version and give feedback which asserts are failing. If you think it is ok to instead have a generic unreached message, I am fine with that as well?

Reply:
That's fair. Maybe then add a comment that you expect them to fail? Also, I forget but do we need a `done()` in case of failed expectations?